pkgs/development/interpreters/perl/wrapper.nix

```diff
@@ -0,0 +1,55 @@
+{ stdenv, perl, buildEnv, makeWrapper
+, extraLibs ? []
+, extraOutputsToInstall ? []
+, postBuild ? ""
+, ignoreCollisions ? false
+, lib
+, requiredPerlModules
+, makeWrapperArgs ? []
+}:
+
+# Create a perl executable that knows about additional packages.
+let
+  env = let
+    paths = requiredPerlModules (extraLibs ++ [ perl ] );
+  in buildEnv {
+    name = "${perl.name}-env";
+
+    inherit paths;
+    inherit ignoreCollisions;
+    extraOutputsToInstall = [ "out" ] ++ extraOutputsToInstall;
+
+    # we create wrapper for the binaries in the different packages
+    postBuild = ''
+
+      . "${makeWrapper}/nix-support/setup-hook"
```

Comment:
You can use `buildInputs = [ makeWrapper ]` instead.

Reply:
The whole file is copy-pasted from ```python```/```lua```.
It has a great potential for optimisation, for example most of the function argumens (in the first lines of the file) are not needed.
I would keep it as is, and then optimize all 3 (python/lua/perl) in sync

Reply:
While it's not really a blocker for this PR, I don't see the point of intentionally using worse style just to keep it consistent, same with the `let self =` thing.

Reply:
the possibilities for optimization here are not limited to these two randomly selected cases and each fix would diverge perl from python and lua making the joint optimisation (for example sharing wrapper.nix) more diffuclt

Reply:
Given the intention is for a follow up PR to optimize and synchronize lua + perl + python I think it makes sense to leave this as is for now. Should we mark this point as resolved?